src/Uno.UI.Runtime.Skia.Wpf/UI/Controls/UnoCompositeWindowHost.cs

```diff
@@ -73,20 +74,25 @@ public UnoWpfWindowHost(UnoWpfWindow wpfWindow, MUX.Window winUIWindow)
 
 		_nativeOverlayLayer = new WpfCanvas();
 		// Transparency doesn't work with the OpenGL renderer, so we have to use the software renderer for the top layer
-		_renderLayer = new RenderingLayerHost(WpfRendererProvider.CreateForHost(this));
+		_renderLayer = new RenderingLayerHost();
 
 		Loaded += WpfHost_Loaded;
 		Unloaded += (_, _) => _backgroundDisposable.Dispose();
 
-		UpdateRendererBackground();
 		_backgroundDisposable.Disposable = _winUIWindow.RegisterBackgroundChangedEvent((_, _) => UpdateRendererBackground());
 	}
 
+	public WpfWindow Window => _wpfWindow;
+
 	public WpfControl RenderLayer => _renderLayer;
 	public WpfControl BottomLayer => _renderLayer;
 
 	private void WpfHost_Loaded(object _, System.Windows.RoutedEventArgs __)
 	{
+		_renderLayer.Renderer = WpfRendererProvider.CreateForHost(this);
```

Comment:
We can get the HWND in wpf after SourceInitialized event. The SourceInitialized be raised before Loaded event.

Reply:
Done 👍